GoogleForCreators/web-stories-wp
# Bugfix/11311 focus border first item

## Context


Properly style focus elements in the context menu

## Summary


Simulates a 'tab' keydown event after custom key binding to open context menu.

## Relevant Technical Choices


We currently use the node module 'focus-visible' - https://www.npmjs.com/package/focus-visible#how-it-works, to control how the focus elements are styled.  The only way to the module runs is when `focus()` is called immediately after `Tab`, `Tab + Shift`, or the arrow keys.  Simulating the 'tab' keydown event after our custom keybinding allows the module to properly shift they styling of element in focus.

## To-do



## User-facing changes


before/after
https://user-images.githubusercontent.com/14057928/167502964-f841907c-0b5c-46c3-b515-a1d094de728b.mp4

## Testing Instructions






This PR can be tested by following these steps:

1. Go to the Story Editor
2. Add a component to the canvas (Text, Shape, etc)
3. Click the component
- Notice the toolbar/context menu pops up
- Use the keyboard shortcut `ctrl+alt+p` to move focus into the context menu
4. One the 'right click' context menu with the keybinding `ctrl+shift+alt+m`

On Staging:
- Notice the first item in the context menu is in focus  (you can press enter to edit the first item in the context menu) but has no focus outline

On Branch:
- Notice the first item is in focus and has a focus outline

## Reviews

### Does this PR have a security-related impact?
No


### Does this PR change what data or activity we track or use?
No


### Does this PR have a legal-related impact?
No


## Checklist




---



Fixes #11311
